src/errors.rs

```diff
@@ -48,4 +50,13 @@ pub enum AggregatorError {
     #[error("bad group sig")]
     /// The aggregate group signature failed to verify
     BadGroupSig,
+    #[error("taproot {0:?}")]
+    /// Taproot error
+    Taproot(TaprootError),
+}
+
+impl From<TaprootError> for AggregatorError {
```

Comment:
I think this block can be replaced if you change 

```rust
   #[error("taproot {0:?}")]
    /// Taproot error
    Taproot(TaprootError),
```
Above to 
```rust
   #[error("taproot {0:?}")]
    /// Taproot error
    Taproot(#[from] TaprootError),
```

Reply:
Actually, I was able to remove `taproot::Error` entirely, since it was mainly used for the `OddR` case.